src/main/scala/util/ShiftReg.scala

```diff
@@ -136,7 +169,8 @@ class SyncResetSynchronizerShiftReg(w: Int = 1, sync: Int = 3, init: Int = 0) ex
 
   override def desiredName = s"SyncResetSynchronizerShiftReg_w${w}_d${sync}_i${init}"
 
-  io.q := ShiftRegInit(io.d, n = sync, init = init.U, name = Some("sync"))
+  val shiftRegs = List.tabulate(w) { i => ShiftRegInit[UInt](io.d(w), n = sync, init = init.U, name = Some(s"sync_${i}")) }
```

Comment:
but won't this not use the new Synchronizer primitive you've created? Maybe we should just delete this class entirely. I don't think we really need to make a sync-reset version of the synchronizer primitive. I'm not sure what in the code base is using this version.

The `SynchronizerShiftReg` is useful though, and should also be replaceable internally by your `SynchronizerPrimitive` but maybe should be renamed explicitly `NoResetSynchronizerShiftReg` and would just be a wrapper around the `AsyncResetSynchronizerShiftReg` that does a `withReset(false.B)`